### PR TITLE
Moves openBLAS and scipy build from script to Dockerfile

### PR DESCRIPTION
**Pull request by fhisamoto**
On failure of one single download in script was necessary to compile the entire OpenBLAS library again. This way the build step is saved in a intermediary image by docker.

### CODE REVIEW

**Comment by rgbkrk** [comment]
This is what I prefer, honestly. Others have complained about image size (and number of layers) though, which is why we've opted for the single script + cleanup.

I'll think on this a bit. You did a really clean layout for the Dockerfile here.

/cc @ogrisel

**Comment by fhisamoto** [comment]
The image size bothers me too. I changed Dockerfile in order to keep each step small, and also was decreased the number of steps.

File: scipystack/Dockerfile [hunk 1]
```diff
@@ -2,19 +2,42 @@ FROM ipython/ipython
 
 MAINTAINER IPython Project <ipython-dev@scipy.org>
 
+RUN apt-get -y update
+RUN apt-get -y install git-core build-essential gfortran python3-dev curl
```
Comment (rgbkrk) [inline scipystack/Dockerfile]: The [ipython/ipython Docker image already includes](https://github.com/ipython/ipython/blob/master/Dockerfile#L22-L30) `git`, `build-essential`, and `python3-dev`.

File: scipystack/Dockerfile [hunk 2]
```diff
@@ -2,19 +2,42 @@ FROM ipython/ipython
...
+ADD config/openblas.conf /etc/ld.so.conf.d/openblas.conf
+RUN git clone -q --branch=master git://github.com/xianyi/OpenBLAS.git
+RUN cd OpenBLAS && \
+    make DYNAMIC_ARCH=1 NO_AFFINITY=1 NUM_THREADS=32 && \
+    make install && \
+    ldconfig && \
+    rm -rf /tmp/OpenBLAS
+
+RUN apt-get build-dep -y python3 python3-numpy python3-scipy python3-matplotlib cython3 python3-h5py && \
+    apt-get install -y build-essential python3-dev && \
```
Comment (rgbkrk) [inline scipystack/Dockerfile]: Another install of `build-essential` and `python3-dev`.

File: scipystack/Dockerfile (hunk 1 above)
Comment (rgbkrk) [inline scipystack/Dockerfile]: These came from the build script we had in place before, but now that things have changed we should be able to clean a few of these up/out.

**Comment by ogrisel** [comment]
maybe the ipython / jupyter docker images could just be based on the `debian:jessie` base image and use the binary packages for `libopenblas-base` / `libopenblas-dev`:

https://packages.debian.org/source/jessie/openblas

Version 0.2.12 is recent enough and the debian package maintainers build openblas with similar options by default (see the `rules` file of the sources of the debian package), in particular:

```
DYNAMIC_ARCH=1 NO_AFFINITY=1 NO_WARMUP=1 NUM_THREADS=64
```

The openblas package in `ubuntu:lastest` is still in 0.2.8 on the other hand. It's a bit old in particular it does not support the dynamic architecture mode that includes the compute kernels for all the supported architectures and select the best at runtime.

**Comment by bollwyvl** [comment]
+1 to Debian base boxes

**Comment by fhisamoto** [comment]
I made changes in Dockerfile to be more cleaner.

**Comment by rgbkrk** [comment]
I'll go ahead and make a PR to ipython/ipython to make our base box be `debian:jessie`. Thanks @ogrisel and @bollwyvl.

**Comment by Carreau** [comment]
What's the status of this ? shoudl we close and reopen if needed later ?

**Comment by Carreau** [comment]
I'm going to close for inactivity, feel fre to reopen.

**Pull request closed without merging**